cppcache/src/DistributedSystemImpl.hpp

```diff
@@ -84,16 +84,17 @@ class _GEODE_EXPORT DistributedSystemImpl {
 
   /** Retrieve the MemberId used to create this Cache. */
   virtual void disconnect();
+
   virtual void connect();
-  std::string m_name;
-  DistributedSystem* m_implementee;
 
+  std::string m_name;
+  DistributedSystem m_implementee;
   DiffieHellman m_dh;
 
   /**
    * @brief constructors
    */
-  DistributedSystemImpl(const char* name, DistributedSystem* implementee);
+  DistributedSystemImpl(std::string name, DistributedSystem implementee);
```

Comment:
This is going to copy the `DistributedSystem`.  You need to keep a pointer here. `DistributedSystemImpl` is owned by `DistributedSystem`, the pointer back to `DistributedSystem` is non-owning reference. See `Cache` and `CacheImpl` pimpl model.